satpy/readers/insat3d_img_l1b_h5.py

```diff
@@ -0,0 +1,197 @@
+"""File handler for Insat 3D L1B data in hdf5 format."""
+from contextlib import suppress
+from datetime import datetime
+from functools import cached_property, lru_cache
+
+import dask.array as da
+import numpy as np
+import xarray as xr
+from datatree import DataTree
+
+from satpy.readers.file_handlers import BaseFileHandler
+
+LUT_SUFFIXES = {"vis": ("RADIANCE", "ALBEDO"),
+                "swir": ("RADIANCE",),
+                "mir": ("RADIANCE", "TEMP"),
+                "tir1": ("RADIANCE", "TEMP"),
+                "tir2": ("RADIANCE", "TEMP"),
+                "wv": ("RADIANCE", "TEMP"),
+                }
+
+CHANNELS_BY_RESOLUTION = {1000: ["vis", "swir"],
+                          4000: ["mir", "tir1", "tir2"],
+                          8000: ["wv"],
+                          }
+
+
+def apply_lut(data, lut):
+    """Apply a lookup table."""
+    return lut[data]
+
+
+def decode_lut_arr(arr, lut):
+    """Decode an array using a lookup table."""
+    dtype = lut.dtype
+    lut_attrs = lut.attrs
+
+    attrs = arr.attrs
+    attrs["units"] = lut_attrs["units"]
+    attrs["long_name"] = lut_attrs["long_name"]
+    new_darr = da.map_blocks(apply_lut, arr.data, lut=np.asanyarray(lut), dtype=dtype)
+    new_arr = xr.DataArray(new_darr, dims=arr.dims, attrs=attrs, coords=arr.coords)
+    new_arr = new_arr.where(arr.data != attrs["_FillValue"])
+    return new_arr
+
+
+def get_lonlat_suffix(resolution):
+    """Get the lonlat variable suffix from the resolution."""
+    if resolution == 1000:
+        lonlat_suffix = "_VIS"
+    elif resolution == 8000:
+        lonlat_suffix = "_WV"
+    else:
+        lonlat_suffix = ""
+    return lonlat_suffix
+
+
+@lru_cache
+def open_dataset(filename, resolution=1000):
+    """Open a dataset for a given resolution."""
+    h5ds = xr.open_dataset(filename, engine="h5netcdf", chunks="auto")
+    h5ds_raw = xr.open_dataset(filename, engine="h5netcdf", chunks="auto", mask_and_scale=False)
+    ds = xr.Dataset()
+    ds.attrs = h5ds.attrs
+    if resolution in [1000, 4000, 8000]:
```

Comment:
Flip this condition and raise the ValueError at the top of the method before you open the dataset at all.

Reply:
Good idea!

Reply:
fixed in [be65fb6](https://github.com/pytroll/satpy/pull/2265/commits/be65fb64e234b9ee39ece54cca2daae0aeb0b8a9)